syntax = "proto3";

package emulation;

// Emulator grpc service is used as an controling point for external emulation
// subsystem which generate the user activity and sends it to our emulation
// network.
service Emulator {
    //
    // SendPayment is used to emulate the activity of one user sending payment
    // to another within the local router network.
    rpc SendPayment (SendPaymentRequest) returns (SendPaymentResponse);

    //
    // OpenChannel is used to emulate that user has opened the channel with the
    // router.
    rpc OpenChannel (OpenChannelRequest) returns (OpenChannelResponse);

    //
    // CloseChannel is used to emulate that user has closed the channel with the
    // router.
    rpc CloseChannel (CloseChannelRequest) returns (CloseChannelResponse);
}

message SendPaymentRequest {
    // Sender is the id of user which sends the payment. If equals zero than we
    // believe that router is the sender.
    uint64 sender = 1;

    // Receiver is the id of user who receive the payment. If equals zero than we
    // believe that router is the receiver.
    uint64 receiver = 2;

    uint64 amount = 4;
}

message SendPaymentResponse {
}

message OpenChannelRequest {
    uint64 user_id = 1;
    uint64 locked_by_user = 2;
    uint64 chan_id = 3;
}

message OpenChannelResponse {
}

message CloseChannelRequest {
    uint64 chan_id = 3;
}

message CloseChannelResponse {
}

